File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnUseCardMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import AbstractCard from "../Cards/AbstractCard";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnUseCardMessage  implements IMessage{
    public static readonly Id = "OnUseCardMessage";
    constructor(public readonly Source:AbstractCreature,public Target:AbstractCreature[],public readonly Card:AbstractCard) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnUseCardMessage.cs
using ModTheSpire.Card;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///使用卡牌时
    public class OnUseCardMessage : IEventHandle
    {
        public AbstractCreature Source;
        public AbstractCreature Target;      
        public AbstractCard Card;

        public OnUseCardMessage(AbstractCreature source, AbstractCreature target, AbstractCard card)
        {
            Source = source;
            Target = target;
            Card = card;
        }
    }
}